### PR TITLE
Feature/replat 604 article templates responsive

### PR DESCRIPTION
**Pull request by jimhunty**
article responsive

This was tested on chrome across the breakpoints set out in the production website against its code.

We are using Styled Components to make this responsive. We are using React Native Stylesheet first and only applying the required styled in Styled Components where needed.

Our aim was to keep everything to do with making the article responsive, in the article package. Keeping all other child packages pure. We have aimed to reduce the number of `.web.js` files.

We wanted to put Styled Components and React Native StyleSheets in the same folder but this upset `React-Native-Web` for unknown reasons.

### CODE REVIEW

File: packages/article-byline/article-byline-proptypes.js [hunk 1]
```diff
@@ -8,10 +8,12 @@ export const articleBylinePropTypes = {
     container: ViewPropTypes.style,
     byline: Text.propTypes.style,
     link: Text.propTypes.style
-  })
+  }),
+  WrapperComponent: PropTypes.func
```
Comment (fourlastor) [inline packages/article-byline/article-byline-proptypes.js]: shouldn't this (and other `WrapperComponent`s) be `PropTypes.element`?

File: packages/article/article.web.js [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+      standfirst,
+      label,
+      byline,
+      publishedTime,
+      publicationName,
+      leadAsset
+    } = articleData;
+    const LeadAssetMedia = ArticlePage.renderLeadAsset(leadAsset);
+    const contentArray = articleData.content.map((i, index) => ({
+      data: i,
```
Comment (fourlastor) [inline packages/article/article.web.js]: maybe just rename `i` to `data`? 😄

File: packages/article/article.web.js [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+      index
+    }));
+    const BodyView = contentArray.map(i => ArticlePage.renderBody(i));
```
Comment (fourlastor) [inline packages/article/article.web.js]: what does `i` represent here?

File: packages/article/article.web.js [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+  static getImageContainer(imageType) {
+    switch (imageType) {
+      case "primary":
+        return PrimaryImg;
+      case "secondary":
+        return SecondaryImg;
+      case "inline":
+        return InlineImg;
+      default:
+        return null;
```
Comment (fourlastor) [inline packages/article/article.web.js]: is this an expected case? or can we throw to catch the error soon in development?

File: packages/article-byline/article-byline-proptypes.js (hunk 1 above)
Comment (jimhunty) [reply packages/article-byline/article-byline-proptypes.js]: This causes proptype errors in React 16 - the suggestion is to now use func.

File: packages/article/article.web.js (hunk 2 above)
Comment (jimhunty) [reply packages/article/article.web.js]: Renamed this,

File: packages/article/article.web.js (hunk 4 above)
Comment (jimhunty) [reply packages/article/article.web.js]: There isnt really a default case so will just choose Primary.

**Comment by aronshamash** [comment]
Couple of things while looking at it integrated into the website (render):
1. Headline font size too small
2. Headline padding needs increasing
3. Article Meta needs to line up with the top of the lead asset

image

**Comment by jimhunty** [comment]
@aronshamash When testing this I see the issue with padding, we are missing the article container for the site. We should be able to resolve this.

The body text has definitely gone strange, but there are no missing fonts.

The pages doesn't scroll either...

I got the following message in my console when running Render:
`[BABEL] Note: The code generator has deoptimised the styling of "{path to site}/Times/cps-content-render/src/article-page.bundle.js" as it exceeds the max of "500KB".`

Im wondering could this be the cause of the last two problems?

**Comment by toboid** [comment]
@jimhunty this error message is raised during the creation of a server-side bundle for server rendering of the article. The way this is done needs to be reviewed however I don't think it's the cause of your issue; its babel saying that it won't trim extra whitespace and line terminators because the file size is >500kb. It's the `compact` option explained on [this](https://babeljs.io/docs/usage/api/) page

**Comment by jimhunty** [comment]
@aronshamash Those issues have now been addressed.

@toboid Thanks for that feedback. My original comment was based on not knowing a few things about `Render`, and therefore we can negate those now.

File: packages/article-byline/article-byline-proptypes.js (hunk 1 above)
Comment (craigbilner) [reply packages/article-byline/article-byline-proptypes.js]: @jimhunty I don't understand `WrapperComponent`, why not just wrap-the-component?

File: packages/article-image/component.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+import { View, StyleSheet } from "react-native";
+import Image from "@times-components/image";
+import Caption from "@times-components/caption";
+
+import {
+  articleImagePropTypes,
+  articleImageDefaultPropTypes
+} from "./article-image-proptypes";
+
+// NOTE: We support only Mobile web, ios and android. Desktop to follow after responsive design
```
Comment (craigbilner) [inline packages/article-image/component.js]: not relevant anymore?

**Comment by craigbilner** [comment]
merged mega PR, raised tech debt issues offline

**Pull request merged**